app/views/student_teams/view.html.erb

```diff
@@ -120,7 +120,7 @@
          <%= text_field 'team', 'name' %>
          <%= submit_tag "Name team" %>
        <% end %>
-     <% elsif !@student.team.full?%>
+     <% elsif !@student.team.full? && @student.assignment.has_topics? && @student.team.topic%>
```

Comment:
Don't think the logic here is correct, should be 

(!@student.team.full? && !@student.assignment.has_topics?) || (!@student.team.full? && @student.assignment.has_topics? && @student.team.topic)